### DISCUSSION TITLE
How to send Response Error 401 from custom KeycloakConfigResolver

### DISCUSSION THREAD

**Discussion by ReginaldoSantos** [keycloak/keycloak#19998] [Q&A]
**Keycloak: 20.0.3**

So, we have these spring boot microservices using Keycloak Authorization Services via keycloak-adapter.

The keycloak-adapter has been extended with a custom KeycloakConfigResolver so we can get user and realm information from a Bearer Token and also support multi-tenancy by building our KeycloakDeployments/AdapterConfigs on the fly.

We are also relying on pure keycloak-adapter servlet-api container security features and thus, we don't have spring-security in place.

It all works well, however we would like to throw a proper HTTP 401 Error when the bearer token is missing in the request, but all our attempts with different kinds of Exceptions (even with org.keycloak.authorization.client.util.HttpResponseException) get wrapped in the container by a HTTP 500 Error.

Is this a bug, missing feature or by design? Is there some other way to achieve the very same behavior?

**Comment by mounir4019** [comment]
Hi,
Are you find a solution?? 
I have same challenge since a week.
